apps/prairielearn/src/pages/instructorAssessments/instructorAssessments.html.tsx

```diff
@@ -196,3 +199,189 @@ export function AssessmentStats({ row }: { row: AssessmentStatsRow }) {
     </td>
   `;
 }
+
+export function PreactInstructorAssessments({
+  resLocals,
+  rows,
+  assessmentIdsNeedingStatsUpdate,
+  csvFilename,
+}: {
+  resLocals: Record<string, any>;
+  rows: AssessmentRow[];
+  assessmentIdsNeedingStatsUpdate: string[];
+  csvFilename: string;
+}) {
+  const { urlPrefix, authz_data, course, __csrf_token } = resLocals;
+
+  return renderHtmlDocument(
+    <html lang="en">
+      <head>
+        <PreactHeadContents resLocals={resLocals} />
+        <script src={compiledScriptPath('instructorAssessmentsClient.ts')}></script>
+        <div
+          dangerouslySetInnerHTML={{
+            __html: EncodedData<StatsUpdateData>(
+              { assessmentIdsNeedingStatsUpdate, urlPrefix },
+              'stats-update-data',
+            ).toString(),
+          }}
+        />
+      </head>
+      <body>
+        <PreactNavbar resLocals={resLocals} />
+        <main id="content" class="container-fluid">
+          <div
+            dangerouslySetInnerHTML={{
+              __html: CourseInstanceSyncErrorsAndWarnings({
+                authz_data,
+                courseInstance: resLocals.course_instance,
+                course,
+                urlPrefix,
+              }).toString(),
+            }}
+          />
```

Comment:
Things like this are hacky workarounds for interoperating with `html` tagged-template literal "components". If we actually want to move forward with this change, we'd probably want to write the component itself with JSX, and then use `preact-render-to-string` to spit out an HTML string that we could use on pages that aren't yet using JSX.